src/adapter/src/client.rs

```diff
@@ -143,11 +144,21 @@ impl Client {
     /// Creates a new session associated with this client for the given user.
     ///
     /// It is the caller's responsibility to have authenticated the user.
-    pub fn new_session(&self, config: SessionConfig) -> Session {
+    pub fn new_session(&self, config: SessionConfig, peer_address: Option<IpAddr>) -> Session {
```

Comment:
I think you probably want to make the `peer_address` part of the `SessionConfig`?